lib/geordi/hint.rb

```diff
@@ -0,0 +1,47 @@
+# Use the methods in this file to hint at other geordi features
+require File.expand_path('settings', __dir__)
+
+module Geordi
+  class Hint
+    class << self
+
+      def did_you_know(hints)
+        hint_probability =  Settings.new.hint_probability
+        default_probability ||= (Util.testing? ? 0 : 10) # Default
```

Comment:
Sorry for the iteration, but now you're `||=`-assigning an uninitialized variable. Please update once more:
```suggestion
        hint_probability = Settings.new.hint_probability
        default_probability = (Util.testing? ? 0 : 10)
```

Reply:
After clarification:
```suggestion
        settings_probability = Settings.new.hint_probability
        default_probability = (Util.testing? ? 0 : 10) # Percent
```